airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/s3/S3StreamCopier.java

```diff
@@ -57,91 +43,90 @@
   protected final JdbcDatabase db;
   private final ExtendedNameTransformer nameTransformer;
   private final SqlOperations sqlOperations;
-  protected final Set<String> s3StagingFiles = new HashSet<>();
-  private final Map<String, StreamTransferManager> multipartUploadManagers = new HashMap<>();
-  private final Map<String, MultiPartOutputStream> outputStreams = new HashMap<>();
-  private final Map<String, CSVPrinter> csvPrinters = new HashMap<>();
-  private final String s3FileName;
+  private final ConfiguredAirbyteStream configuredAirbyteStream;
+  private final Timestamp uploadTime;
   protected final String stagingFolder;
-  private final StagingFilenameGenerator filenameGenerator;
+  protected final Map<String, S3Writer> stagingWritersByFile = new HashMap<>();
 
+
+  // The number of batches of records that will be inserted into each file.
+  private final int maxPartsPerFile;
+  // The number of batches inserted into the current file.
+  private int partsAddedToCurrentFile;
+  private String currentFile;
+
+  /**
+   * @param maxPartsPerFile The number of "chunks" of requests to add into each file. Each chunk can be up to 256 MiB (see
```

Comment:
parts are not all held in memory at the same time right? if so worth mentioning in the comment

Reply:
I think (because we don't close the uploads on the fly) each part probably has some data in-memory until `closeStagingUploader` is called. But certainly not the whole part. How does something like this sound:

> Each new file consumes some memory, as we don't close the output stream until {@link #closeStagingUploader} is called. However, the file is streamed into S3 dynamically, and is never fully-held in-memory.